utils_nlp/models/transformers/common.py

```diff
@@ -47,11 +49,48 @@ def get_device(device, num_gpus, local_rank):
 
 
 class Transformer:
-    def __init__(self, model_class, model_name="bert-base-cased", num_labels=2, cache_dir="."):
+    def __init__(
+        self,
+        model_class,
+        model_name="bert-base-cased",
+        num_labels=2,
+        cache_dir=".",
+        load_model_from_dir=None,
+        config_class=None,
+    ):
         self.model_name = model_name
-        self.model = model_class[model_name].from_pretrained(
-            model_name, cache_dir=cache_dir, num_labels=num_labels
-        )
+        self.cache_dir = cache_dir
+        self.load_model_from_dir = load_model_from_dir
+        if load_model_from_dir is None:
```

Comment:
doesnt from_pretrained() load from cache_dir anyway, if the file exists?

Reply:
It's a bit tricky. This argument is intended to load model that is saved by save_pretrained(), usually a fine tuned model. The model file name is hard-coded as pytorch_model.bin in the save_pretrained() function. 
1. When you call from_pretrained(model_name, cache_dir=cache_dir, num_labels=num_labels), the function tries to find the specific model file name, e.g. bert-large-cased-whole-word_masking, corresponding to the model_name argument and loads it. If it couldn't find it, it will download it from s3. 
2. When you call  from_pretrained(cache_dir, num_labels=num_labels), it loads the model file saved by save_pretrained().
I thought about changing load_model_from_dir to a boolean argument load_model_from_cache, and do #1 if it's this flag is False and do # 2 if this flag is True.  But it could confuse people, because even the flag is False, for #1 above, the model is loaded from cache_dir if it already exists.
I do find the config is not necessary and will have a PR to remove it soon.
This is the best I can think of. We can discuss to see if there is a better way to handle this.